### ISSUE TITLE
QTree &  tick-strategy="leaf" & function setTicked

### ISSUE THREAD

**Issue by turigeza** [quasarframework/quasar#7052]
**Is your feature request related to a problem? Please describe.**
When I use QTree in combination with  tick-strategy="leaf" I am allowed to set parent nodes as ticked. This bugs me as this can not happen when you simply click the parent node's checkbox. Also if this is allowed one should also be able to see that the node status is ticked. See the pen how it ends up looking odd. 

Model shows that "Satisfied customers" is ticked.
QTree doesn't.
https://codepen.io/turigeza/pen/PoPdMbK?editors=1010

**Describe the solution you'd like**
I would like the to be able to programmatically tick parent nodes and leave it to QTree to decide if the children needed to be ticked or unticked according to the `tick-strategy`. Just like if I triggered a 'click' on the node's checkbox.

Otherwise I am forced to reimplement the tick strategy behaviour. Now changin the function setTicked would be breaking change so perhaps a new function triggerTick() or something on that line would be great. Thank you for the great work you all do : )

**Additional context**
<img width="590" alt="Screenshot 2020-05-15 at 18 15 19" src="https://user-images.githubusercontent.com/9335740/82077836-fb82a680-96d7-11ea-986f-42a64885f69b.png">

**Comment by pdanpdan** [comment]
What I do not understand is what you need different from the `strict` strategy, can you please explain?

**Comment by turigeza** [comment]
Sort of ish ... : ) but not quite ...

I would like to be able to "trigger a click" on any of the node checkboxes. What happens after that is entirely up to QTree and it's `tick-strategy`s.  

But I don't want to do this the old fashioned way like triggering a click on the dom element. 

Why tick strategies are need ? That is a different question. So a typical case when you are classifying plants. 
Yu have their ..
CLASS
SUBCLASS
SUPERORDER
ORDER
FAMILY
SUBFAMILY
TRIBE
GENUS
SPECIES
SUBSPECIES
VARIETY
 
Now if I was tagging a plant with it's VARIETY I would also like to tag it with its SUBSPECIES, SPECIES, GENUS and so on. You would not want the user to have to click every single one of those as well. 
Now this tick strategy is actually not provided by QTree but that is a different matter. I could implement this once I get access to `get node parents ` in this list of feature request. 
QTree feature requests summary #6959

**Comment by turigeza** [comment]
Actually it isn't that easy to implement my own tick strategy after all because `@update:ticked -> function(target)` does not give me the node which was ticked. Instead it provides me an array of all ticked items. : ( 

I would like update `@update:ticked` to get the node which was ticked as a second argument and if it was ticked or unticked as the third.

`@update:ticked -> function(target, tickedNode, onOff)`

Shall I submit a separate feature request for this ?

**Comment by pdanpdan** [comment]
No, I'll put it in the list. This request makes sense, buy to be honest the first one looks pointless and useless for me. We try to work from data to UI, so the normal way is to fill the array of ticked elements.

**Comment by turigeza** [comment]
> We try to work from data to UI, so the normal way is to fill the array of ticked elements.
Sure thing. So am I. And this is just how Vue is and I wouldn't want it any different either. 

When I say "trigger a click" I mean I would like to achieve the effect of that without literally triggering a click. 

The problem is now that I have no way of problematically ticking a node unless I reverse engineer your ticking strategies. Or I have to discard your ticking strategies and implement my own. Neither is ideal btw.

If we go back to the original screenshot and pen.
https://codepen.io/turigeza/pen/PoPdMbK?editors=1010

This was my expected outcome. 
<img width="582" alt="Screenshot 2020-05-16 at 10 43 25" src="https://user-images.githubusercontent.com/9335740/82116455-e6eaf080-9761-11ea-89c4-44b02d6eff1f.png">

But yet this is what I get.
<img width="590" alt="Screenshot 2020-05-15 at 18 15 19" src="https://user-images.githubusercontent.com/9335740/82116474-0550ec00-9762-11ea-9101-3bad19b03bfc.png">

In other words the function you provide. `setTicked ` ignores the ticking strategy rules  and you can freely tick any elements you would not be able to do so otherwise.

You would never be able to tick `Satisfied customers` here. 
https://codepen.io/pen/?&editable=true&editors=101=https%3A%2F%2Fquasar.dev%2Fvue-components%2Ftree

IMO because of that the function `setTicked`  is not much use to me. But it's there so might as well just leave it. 

However I would like a version of this function which plays well with your ticking strategies regardless of the strategy used.

**Comment by turigeza** [comment]
Hang on just a minute sorry I got something wrong with the screenshots.

**Comment by xnextx** [comment]
I have exactly the same problem :( Is there a solution for this? I would like to get the actual selected tree, not just the last item in the list.

**Comment by Steeve55** [comment]
I need it also please.

**Comment by CultivateCreate** [comment]
Any update on this? would be useful to show the hierarchical structure of ticked nodes (including the parents...)

**Comment by christian-beckmann** [comment]
@pdanpdan How can you close this ticket, but never really worked on it. This feature still is not present, someone called it "leafParent" here https://github.com/quasarframework/quasar/issues/2508. 
It would be great also to have the parent/root key in ticked.
